### ISSUE TITLE
Performance regression in finite difference benchmark

### ISSUE THREAD

**Issue by romeric** [romeric/Fastor#48]
The `fseq` variant used to be much faster than the scalar code and now it is over 5 times slower

**Comment by romeric** [comment]
Clang does not seem to be affected however GCC is affected badly by this issue.

**Comment by romeric** [comment]
The issue is in `FASTOR_ASSERT` in binary_arithmetic_ops. If this is changed to `assert` then the issue is resolved.

with FASTOR_ASSERT
~~~
./out_loops.exe 100
 Relative error is:  9.99919e-07 
 Number of iterations:  15231 
 Elapsed time is: 0.189134 seconds 
./out_seq_alias.exe 100
 Relative error is:  9.99919e-07 
 Number of iterations:  15231 
 Elapsed time is: 2.88382 seconds 
./out_seq_noalias.exe 100
 Relative error is:  9.99919e-07 
 Number of iterations:  15231 
 Elapsed time is: 2.82128 seconds 
./out_iseq.exe 100
 Relative error is:  9.99919e-07 
 Number of iterations:  15231 
 Elapsed time is: 3.36344 seconds 
./out_fseq.exe 100
 Relative error is:  9.99919e-07 
 Number of iterations:  15231 
 Elapsed time is: 2.81872 seconds 
./out_seq_alias_vec.exe 100
 Relative error is:  9.99919e-07 
 Number of iterations:  15231 
 Elapsed time is: 2.87738 seconds 
./out_seq_noalias_vec.exe 100
 Relative error is:  9.99919e-07 
 Number of iterations:  15231 
 Elapsed time is: 2.86499 seconds 
./out_iseq_vec.exe 100
 Relative error is:  9.99919e-07 
 Number of iterations:  15231 
 Elapsed time is: 3.27621 seconds 
./out_fseq_vec.exe 100
 Relative error is:  9.99919e-07 
 Number of iterations:  15231 
 Elapsed time is: 2.82649 seconds 
~~~

With assert
~~~
./out_loops.exe 100
 Relative error is:  9.99919e-07 
 Number of iterations:  15231 
 Elapsed time is: 0.192972 seconds 
./out_seq_alias.exe 100
 Relative error is:  9.99919e-07 
 Number of iterations:  15231 
 Elapsed time is: 0.212822 seconds 
./out_seq_noalias.exe 100
 Relative error is:  9.99919e-07 
 Number of iterations:  15231 
 Elapsed time is: 0.15528 seconds 
./out_iseq.exe 100
 Relative error is:  9.99919e-07 
 Number of iterations:  15231 
 Elapsed time is: 0.589573 seconds 
./out_fseq.exe 100
 Relative error is:  9.99919e-07 
 Number of iterations:  15231 
 Elapsed time is: 0.151906 seconds 
./out_seq_alias_vec.exe 100
 Relative error is:  9.99919e-07 
 Number of iterations:  15231 
 Elapsed time is: 0.203932 seconds 
./out_seq_noalias_vec.exe 100
 Relative error is:  9.99919e-07 
 Number of iterations:  15231 
 Elapsed time is: 0.153098 seconds 
./out_iseq_vec.exe 100
 Relative error is:  9.99919e-07 
 Number of iterations:  15231 
 Elapsed time is: 0.586355 seconds 
./out_fseq_vec.exe 100
 Relative error is:  9.99919e-07 
 Number of iterations:  15231 
 Elapsed time is: 0.152321 seconds 
~~~

**Comment by romeric** [comment]
Fixed in 487bfbc.
